### PR TITLE
Renames Message to Block

### PR DESCRIPTION
**Pull request by luca-moser**
At the instances where we talk about `Message` as the core protocol model but not where it is used as a raw json message or when as an input to cryptographic functions.

We do this to better align to IOTA 2.0 naming

### CODE REVIEW

File: block.go [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+	}
+	*m = *seri.(*Block)
+	return nil
+}
+
+// jsonBlock defines the JSON representation of a Block.
+type jsonBlock struct {
+	ProtocolVersion int `json:"protocolVersion"`
+	// The hex encoded IDs of the referenced parent blocks.
+	Parents []string `json:"parentBlockIds"`
```
Comment (alexsporn) [inline block.go:231]: do we maybe want to just call it parents in JSON too?

File: milestone.go [hunk 2]
```diff
@@ -491,7 +491,7 @@ type jsonMilestone struct {
 	Timestamp           int                `json:"timestamp"`
 	ProtocolVersion     int                `json:"protocolVersion"`
 	PreviousMilestoneID string             `json:"previousMilestoneId"`
-	Parents             []string           `json:"parentMessageIds"`
+	Parents             []string           `json:"parentBlockIds"`
```
Comment (alexsporn) [inline milestone.go:494]: Here again, maybe just call it `parents`?

File: block.go (hunk 1 above)
Comment (luca-moser) [reply block.go:231]: sounds good to me

Comment (thibault-martinez) [reply block.go]: Should we do the same for the API block metadata response ?

Comment (alexsporn) [reply block.go]: I renamed them

**Pull request merged**